### PR TITLE
[nexus][sled-agent] Try more to load firewall rules during service bringup

### PR DESCRIPTION
**Pull request by smklein**
Fixes https://github.com/oxidecomputer/omicron/issues/5053 , with this solution implemented:

> Nexus Internal Endpoint to Ask for FW Rules, Triggered After Individual Zones Launch. Rather than waiting for "all zones to launch", the Sled Agent could just re-apply firewall rules after any zones that need FW rules come up.

I used a bit of a brute-force solution here -- "just do it each time we try to launch a set of zones". This could certainly be more fine-grained, but it gets the job done.

### CODE REVIEW

File: sled-agent/src/sled_agent.rs [hunk 1]
```diff
@@ -602,11 +605,24 @@ impl SledAgent {
...
+                // partially load some services.
+                let firewall_result = self
+                    .request_firewall_update()
                     .await
-                    .map_err(|err| BackoffError::transient(err))
+                    .map_err(|err| BackoffError::transient(err));
+
+                // Only complete if we have loaded all services and firewall
+                // rules successfully.
+                load_services_result.and(firewall_result)
```
Comment (smklein) [inline sled-agent/src/sled_agent.rs:625]: By doing this operations back-to-back, we basically "load as many firewall rules as we can" for a set of services.

I could make this trigger more individually for "services which need firewall rules" (e.g., the request is only made after **adding** a Nexus / External DNS / boundary NTP zone), but felt that this was more "obviously not missing anything".

I could push down the triggers into the `ServiceManager` if folks felt strongly about this.

Comment (luqmana) [reply sled-agent/src/sled_agent.rs:625]: Hmm, like this we'll continue to hit the endpoint if we get stuck in the disk error loop right? Wonder if cause any issues constantly trying to update fw rules. Oh but there's the backoff in the retry loop. So, should be fine

**Review by luqmana — APPROVED** [review]
Thanks for putting this together so quick Sean!

**Review by FelixMcFelix — APPROVED** [review]
Thanks for the very quick turnaround! This looks about right to me from what we talked over.

**Pull request merged**